src/run.js

```diff
@@ -73,10 +74,25 @@ const loadConfig = testPath =>
 
 // Run tests on a given file
 const run = ({ testPath, config, globalConfig }) => {
-  const schema = require(testPath)
+  const { transform } = config
+
+  let schema
+  let fileContent = fs.readFileSync(testPath, { encoding: 'utf-8' })
+
+  if (
+    !config.transformIgnorePatterns
+      .map(x => new RegExp(x))
+      .find(x => x.test(testPath))
+  ) {
+    const transformer = transform.find(x => new RegExp(x[0]).test(testPath))
+
+    if (transformer) {
+      const transformerObj = require(transformer[1])
+      fileContent = transformerObj.process(fileContent, testPath, config)
+    }
+  }
 
-  // Avoid caching for `--watch` mode
-  delete delete require.cache[require.resolve(testPath)]
+  schema = nodeEval(fileContent, testPath)
```

Comment:
I feel like we should be using the ScriptTransformer from `@jest/transform` which seems to deal with the ignore paths, permutations of how it can be configured, caching and joining of regexes, caching of transformed source.  Unfortunately it's really unclear to me how we do that (or even if we need to; if we specify the transforms does jest already handle this for us or do we need to explicitly handle it like this in custom runners?)

@SimenB is there any documentation available on how to support transforms inside custom runners available?  It might be a useful addition to the docs for https://github.com/jest-community/create-jest-runner.

Reply:
I don't think we have any docs for it - I haven't really thought about this use case. This is untested, but should work

```js
const { readFileSync } = require("fs");
const { ScriptTransformer } = require("@jest/transform");

const run = ({ testPath, config, globalConfig }) => {
  const content = readFileSync(testPath, "utf8");

  const scriptTransformer = new ScriptTransformer(config);

  // if file is ignored by transformer, `null` is returned by `transformSource`
  const transformedContent =
    scriptTransformer.transformSource(testPath, content, false).code || content;

  // use/eval transformedContent however you'd like
};
```

Happy to take a PR documenting this 👍

Reply:
Thanks for the comments @elyobo, I think most of the issues come from vscode formatting the files on save. Do you have an editor config or .prettierrc that I could use to format everything properly?